### PR TITLE
refactor new user form to use react hook form

### LINKED ISSUE #5824: [Add User]  Form not using proper element and error does not display for Testing facility access field
## Area or Component

Settings Page -> Manage users tab -> add user 

## Quick Summary

The add user form is using a div element as a wrapper. This form also does not follow the pattern of having the submit button enabled and displaying validation errors on submit, and because of this the error message for the required field Testing facility access never displays. Also it can be confusing for the user why the button remains disabled as we follow other patterns across the app (button enable and error on submit).

### How to Replicate

Check the DOM for the add user form and also verify that the button never enables and there is no way to show error for required field `Testing facility access`.

## Action Requested

Wrap the add user form in a `<form>` tag. 
Switch the submission pattern to display errors on submit.
Add errors for the required field `Testing facility access`.

Stretch migrate this form to reach-hook-forms

## Screenshots (optional)

Screenshot 2023-05-16 at 3 45 16 PM

### PR DESCRIPTION
**Pull request by BobanL**
# FRONTEND PULL REQUEST

## Related Issue

- Resolves #5824 

## Changes Proposed

- Refactor CreateUserForm to use react-hook-form
  - Replace `Input` with `TextInput` since it's already been modified to support react-hook-forms
  - Map form data to the `SettingsUser`
- Remove unused code from `UserSettingsFacilityForm`

## Additional Information

- React hook forms does not work well with checkboxes when the `checked` property is defaulted to false, so the code was refactored to remove that since undefined is also falsy

## Testing

- Test validation for all fields for creating a user
- Test creating a user works 
- Test modifying a user's facilities or role still works
- Deployed in [Dev](https://dev.simplereport.gov/)

### CODE REVIEW

File: frontend/src/app/Settings/Users/UserFacilitiesSettings.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+}) => {
+  const isAdmin = formValues.role === "ADMIN";
+  const facilityAccessDescription = isAdmin
+    ? "Admins have access to all facilities"
+    : "All users must have access to at least one facility";
+  allFacilities.sort(alphabeticalFacilitySort);
+
+  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value, checked } = e.target;
+    if (value === "ALL_FACILITIES") {
```
Comment (fzhao99) [inline frontend/src/app/Settings/Users/UserFacilitiesSettings.tsx:36]: nit: could we un-nest these logical checks and just put them all in the top row?

**Review by emyl3 — CHANGES_REQUESTED** [review]
One edge case I noticed,
- get in a state where the "At least one facility must be selected" error appears
- then select "Access Level" to be "Admin"
- it will check all the facilities and disable the facility selection but the error remains
Screenshot 2023-06-22 at 11 34 26

**Review by emyl3 — APPROVED** [review]
LGTM! Tested on dev! Thank you, Boban!

**Review by fzhao99 — APPROVED** [review]

**Review by johanna-skylight — APPROVED** [review]
🚀

**Pull request merged**